src/ops/discard.rs

```diff
@@ -10,20 +10,22 @@ impl OpTrait for Discard {
     fn get_action(&self, target: Option<&TargetData>) -> Option<Action> {
         let action = match target.cloned() {
             Some(TargetData::Branch(r)) => cmd_arg(discard_branch, r.into()),
-            Some(TargetData::File(f)) => Rc::new(move |state: &mut State, _term: &mut Term| {
-                let path = PathBuf::from_iter([
-                    state.repo.workdir().expect("No workdir").to_path_buf(),
-                    f.clone(),
-                ]);
-                std::fs::remove_file(path)?;
-                state.screen_mut().update()
-            }),
+            Some(TargetData::File(f)) => cmd_arg(clean_file_cmd, f.into()),
             Some(TargetData::Delta(d)) => {
-                if d.old_file == d.new_file {
-                    cmd_arg(checkout_file_cmd, d.old_file.into())
-                } else {
-                    // TODO Discard file move
-                    return None;
+                match d.status {
+                    git2::Delta::Added => Rc::new(move |state: &mut State, term: &mut Term| {
+                        let file = d.new_file.as_os_str();
+                        state.run_cmd(term, &[], unstage_file_cmd(file))?;
+                        state.run_cmd(term, &[], clean_file_cmd(file))
+                    }),
+                    _ => {
+                        if d.old_file == d.new_file {
+                            cmd_arg(checkout_file_cmd, d.old_file.into())
+                        } else {
+                            // TODO Discard file move
```

Comment:
Optional: I think this old_file == new_file check and TODO can just be removed. I can't recall it ever happening, and should now be covered by your match on line 15 anyway!

Reply:
Done! Is it intentional that it shows moved file as a pair of added-deleted?

Reply:
I think there must be some configuration to do move detection with libgit2.
It doesn't seem to happen out of default. Just a matter of implementing/fixing